### ISSUE TITLE
Demo code

### ISSUE THREAD

**Issue by YangJae96** [ChelsieLei/EZ-HOI#1]
Hi. Thanks for your great work!

Is there a demo code that we can run the model in custom images?!

If so, could you please upload the code.

Thanks in advance.

**Comment by ChelsieLei** [comment]
Hi, 

Thanks for your interest in our work. 
We don't have the demo code, but I think you can replace the dataset path with your own image path. Remember to set up the pre-extracted CLIP image features.
```
python CLIP_hicodet_extract.py
```
Moreover, we provided the visualization code for the prediction by adding the "--vis_img" in the script. "--vis_img_path" can be used to change the result saving path.

**Comment by YangJae96** [comment]
Thank you for your reply.

1. You mean I should reproduce the **main_tip_finetune.py** code by using the argument in **hico_test_vitB_zs.sh**, right?
Should I still load the dataset for model configuration, even when I only want to input custom images?

2. Do I still need to extract hicodet CLIP features by CLIP_hicodet_extract.py, when I only want to input custom images?

Could I get some help?

**Comment by ChelsieLei** [comment]
Hi, 
For your first question, I think you don't need to load the dataset for custom image testing. You only need to change the test data path to your images. Then follow the argument in __hico_text_vitB_zs.sh__. 

For your second question, you don't need to extract the HICO-DET CLIP features. Please modify the image path in __CLIP_hicodet_extract.py__ to your own image path and pre-extract the CLIP image features for your own images. 
In our pipeline, our model requires the input image's CLIP feature for processing, so to accelerate the training and testing, we pre-extracted both training and testing images to obtain the related CLIP features.

**Comment by Cherryjingyao** [comment]
I also try to test on my own images.But I found that using  hico_test_vitB_zs.sh  will load train_dataset 
<img width="819" alt="image" src="https://github.com/user-attachments/assets/f16dcc6f-3d4a-40a0-b807-a3c4cea918e2">  so how to inference on my own image

**Comment by ChelsieLei** [comment]
Hi,

You can comment out the [training dataset](https://github.com/ChelsieLei/EZ-HOI/blob/main/main_tip_finetune.py#L186) and  [training data loader](https://github.com/ChelsieLei/EZ-HOI/blob/main/main_tip_finetune.py#L206) and other code related to the training data, as it won't affect the evaluation. 
Some parameters are loaded together with the training dataset, as [here](https://github.com/ChelsieLei/EZ-HOI/blob/main/main_tip_finetune.py#L251C9-L251C38), but you can obtain the same parameters from test dataset, too.

**Comment by Cherryjingyao** [comment]
想问下这个object_to_verb 是指什么，和instance.json 里的"correspondence"有什么关系。为什么test还需要这个标签
<img width="790" alt="image" src="https://github.com/user-attachments/assets/b609b4ea-b693-4e58-90ad-faac6045517d">

**Comment by ChelsieLei** [comment]
Hi, 

This "object_to_target" represents all possible object and verb combinations. Not all combinations between objects and actions are valid in the HICO-DET setting, e.g., eat bicycle. "object_to_target" filters out invalid predictions.

The "correspondence" in "instance.json" includes all 600 HOI classes in the HICO-DET dataset and each class's related action label and object label. Therefore, the information from "object_to_target" can be extracted from the "correspondence".

**Comment by Cherryjingyao** [comment]
can you provide a  demo for one image  inference ?  It's hard to run the test script 😢

**Comment by ChelsieLei** [comment]
Hi,

We are very busy at the moment and will be for the coming weeks, but we will get back to you once things settle down.

**Comment by ChelsieLei** [comment]
Hi,

We have updated the inference code for one image. You can directly run 
```
bash script/inference.sh
```
Remember to add your image path and the pre-trained model path.

We observed that, although using an object detector pre-trained on HICO-DET is a common practice in HOI detection, it may struggle to detect objects effectively in out-of-distribution scenarios, potentially hindering HOI detection performance. Since the object detector is not the primary focus of our work, users aiming for improved results can explore alternative object detectors and re-train our model accordingly.
